robottelo/cli/factory.py

```diff
@@ -4316,3 +4317,71 @@ def virt_who_hypervisor_config(
+                                                comma should be quoted or escaped with backslash.
+                                               JSON is acceptable and preferred way for complex
+                                               parameters
+     --locations LOCATION_NAMES                Comma separated list of values. Values containing
+                                                comma should be quoted or escaped with backslash.
+                                               JSON is acceptable and preferred way for complex
+                                               parameters
+     --name NAME                               The HTTP Proxy name
+     --organization ORGANIZATION_NAME          Organization name
+     --organization-id ORGANIZATION_ID         Organization ID
+     --organization-ids ORGANIZATION_IDS       REPLACE organizations with given ids.
+                                               Comma separated list of values. Values containing
+                                               comma should be quoted or escaped with backslash.
+                                               JSON is acceptable and preferred way for complex
+                                               parameters
+     --organization-title ORGANIZATION_TITLE   Organization title
+     --organization-titles ORGANIZATION_TITLES Comma separated list of values. Values containing
+                                                comma should be quoted or escaped with backslash.
+                                               JSON is acceptable and preferred way for complex
+                                               parameters
+     --organizations ORGANIZATION_NAMES        Comma separated list of values. Values containing
+                                                comma should be quoted or escaped with backslash.
+                                               JSON is acceptable and preferred way for complex
+                                               parameters
+     --password PASSWORD                       Password used to authenticate with the HTTP Proxy
+     --url URL                                 URL of the HTTP Proxy
+     --username USERNAME                       Username used to authenticate with the HTTP Proxy
+     -h, --help                                Print help
+    """
+    # Assigning default values for attributes
+    args = {
+        'location': None,
+        'location-id': None,
+        'location-title': None,
+        'locations': None,
+        'location-ids': None,
+        'location-titles': None,
+        'name': gen_string('alpha', 15),
+        'organization': None,
+        'organization-id': None,
+        'organization-title': None,
+        'organizations': None,
+        'organization-ids': None,
+        'organization-titles': None,
+        'password': None,
+        'url': 'https://{}'.format(gen_string('alpha', 15)),
```

Comment:
```suggestion
        'url': '{}:{}'.format(gen_url(scheme='https'), gen_integer(min_value=10, max_value=9999)),
```
1. https url could genrate using `gen_url`
2. Add the port as well with URL.

Reply:
Thanks, updated PR.